### ISSUE TITLE
Server::set_payload_max_length not working

### ISSUE THREAD

**Issue by sweihub** [yhirose/cpp-httplib#1699]
Hi

I am trying to post a JSON file which is greater than 8K, even after I called the `Server::set_payload_max_length`,  I got an error `HTTP/1.1 413 Payload Too Large`, after debugging, I find the code dose not respect the config, can you fix? Thanks.

https://github.com/yhirose/cpp-httplib/blob/e2813d9d4d4a3cb7ee5b1902cf44a0b9529c921e/httplib.h#L5998-L6001

Ref: https://github.com/yhirose/cpp-httplib/issues/144

**Comment by yhirose** [comment]
Here is an running example. Hope it helps!
https://github.com/yhirose/cpp-httplib/blob/e2813d9d4d4a3cb7ee5b1902cf44a0b9529c921e/test/test.cc#L4798-L4849

**Comment by fruityloops1** [comment]
> Here is an running example. Hope it helps!
> 
> https://github.com/yhirose/cpp-httplib/blob/e2813d9d4d4a3cb7ee5b1902cf44a0b9529c921e/test/test.cc#L4798-L4849

I'm having the same issue with >8192 sized requests being ignored even if i call set_payload_max_length with a value higher than that. How does this example explain that?

**Comment by Gondlar** [comment]
I have opened a new issue about this since the root cause has been overlooked here; see yhirose/cpp-httplib#1737.

TLDR: Set `CPPHTTPLIB_FORM_URL_ENCODED_PAYLOAD_MAX_LENGTH` to a higher value in your compiler options. As of yet, `Server::set_payload_max_length` can only limit the payload size below that value, not raise it higher. The test mentioned above shows that this restricting behavior works but does not address raising the limit higher.
